protocol/x/revshare/keeper/msg_set_marketmapper_rev_share_details.go

```diff
@@ -0,0 +1,32 @@
+package keeper
+
+import (
+	"context"
+
+	errorsmod "cosmossdk.io/errors"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/dydxprotocol/v4-chain/protocol/x/revshare/types"
+)
+
+func (k msgServer) SetMarketMapperRevShareDetailsForMarket(
+	goCtx context.Context,
+	msg *types.MsgSetMarketMapperRevShareDetailsForMarket,
+) (*types.MsgSetMarketMapperRevShareDetailsForMarketResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// Check if sender is authorized to set revenue share
+	if !k.HasAuthority(msg.Authority) {
+		return nil, errorsmod.Wrapf(
+			govtypes.ErrInvalidSigner,
+			"invalid authority %s",
+			msg.Authority,
+		)
+	}
+
```

Comment:
Need validation to check if the provided marketId is valid.

You can do this by fetching the market from the prices keeper and checking if it exists

Reply:
Chatted offline and decided not to have this validation as we assume governance is a strong safety check. Having this validation creates a circular dependency between x/prices and x/revshare

Reply:
Call this out explicitly in a comment.